boto3/docs/resource.py

```diff
@@ -312,6 +312,17 @@ def _add_waiters(self, section):
             documenter.document_resource_waiters(section)
             self._add_overview_of_member_type(section, 'waiters')
 
+    def _add_feature_freeze_note(self, section):
+        section = section.add_new_section('feature-freeze')
+        section.style.start_note()
+        section.write(
+            ("The AWS Python SDK team does not intend to add new features to the resources "
+            "interface in boto3. Existing interfaces will continue to operate during "
+            "boto3's lifecycle. Customers can find access to newer service features through "
+            f"the :doc:`client interface <../../{self._service_name}>`.")
+        )
+        section.style.end_note()
+
```

Comment:
Is this generated for every resource? I've forgotten if we've discussed this previously. We may want to link back to the main resource page with that message rather than adding it everywhere.

Reply:
Yes, this is generated at the top of every resource page. Not sure what you mean by:

> link back to the main resource page with that message rather than adding it everywhere.

Would we alter the message saying something like "Please see <Resources Page> for the current guidance on using boto3 Resources"?

Reply:
Yep, something like that would be good.